pandas/core/construction.py

```diff
@@ -298,6 +309,12 @@ def array(
             # timedelta, timedelta64
             return TimedeltaArray._from_sequence(data, copy=copy)
 
+        elif inferred_dtype in {"string", "mixed-string"}:
+            return StringArray._from_sequence(data, copy=copy)
+
+        elif inferred_dtype in {"integer", "mixed-integer"}:
+            return IntegerArray._from_sequence(data, copy=copy)
```

Comment:
IIUC the IntegerArray part doesnt depend on the lib.inferr_dtype change?

Reply:
We already infer `[1, None]` as `mixed-integer`. The only change necessary was handling that case here, rather than falling through to PandasArray.

Reply:
To repeat here what I just commented on the issue: shouldn't we be inferrring the dtype with `skipna=True`? Then [1, None] will also give 'integer' and not 'mixed-integer'